src/read.rs

```diff
@@ -358,7 +366,11 @@ impl<'a> SliceRead<'a> {
                 }
                 _ => {
                     if validate {
-                        return error(self, ErrorCode::InvalidUnicodeCodePoint);
+                        if self.slice[self.index] <= b'\x1F' {
+                            return error(self, ErrorCode::ControlCharacterWhileParsingString);
+                        } else {
+                            return error(self, ErrorCode::InvalidUnicodeCodePoint);
```

Comment:
What input would hit this line?

Reply:
Hmm... I believe `&[b'"', b'\n', b'"']` will hit that line.  I added two entries in `test_parse_slice_err` for this.

Beyond that, I don't really understand when `SliceRead` would be used instead of `IoRead`.  I updated all sites with `InvalidUnicodeCodePoint` except those sites dealing with decoding escaped unicode code points.  I may have gone overboard, so if this change isn't appropriate, I'll take it out.

Reply:
I mean specifically this line. The new test cases are hitting the return two lines up from this one.

Reply:
Oh, good catch. I didn't realize the `_` arm of the `match` was only handling the `CT` escapes, since backslash and quote were handled by the other arms.  No input hits that line.  Not there, and not for the other 3 sites I changed.

I originally assumed there were other unicode-related cases that would trigger the `_` arm and threw in the `if-else` to catch those.

Patch simplified to remove the extraneous `if-else`.